cms/djangoapps/contentstore/courseware_index.py

```diff
@@ -175,8 +181,31 @@ def index_item(item, skip_index=False, groups_usage_info=None):
                 return
 
             item_content_groups = None
+
+            if item.category == "split_test":
+                user_partitions = item.user_partitions
+                split_partitions = []
+                if user_partitions:
+                    for user_partition in user_partitions:
+                        if user_partition.scheme.name == RANDOM_SCHEME:
+                            split_partitions.append(user_partition)
+                for vertical in item.get_children():
```

Comment:
Note that the children of a split test do not have to be verticals. There are examples of MIT-built XML courses that use split tests around other kinds of children (I believe chapters). The key thing is that there is a mapping from group to child in the split_test xblock, and that this group applies to the child and all of its descendants.